functests/1_performance/performance.go

```diff
@@ -340,8 +342,134 @@ var _ = Describe("[rfe_id:27368][performance]", func() {
 			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("cannot find MachineConfig object %s", initialKey.Name))
 		})
 	})
+
+	Context("Verify API Conversions", func() {
+		It("[test_id:35887] Verifies v1 <-> v1alpha1 conversions", func() {
+
+			By("Checking v1 -> v1alpha1 conversion")
+			v1alpha1Profile := &performancev1alpha1.PerformanceProfile{}
+			key := types.NamespacedName{
+				Name:      profile.Name,
+				Namespace: profile.Namespace,
+			}
+
+			err := testclient.GetWithRetry(context.TODO(), key, v1alpha1Profile)
+			ExpectWithOffset(1, err).ToNot(HaveOccurred(), "Failed getting v1alpha1Profile")
+			Expect(verifyV1alpha1Conversion(v1alpha1Profile, profile)).ToNot(HaveOccurred())
+
+			By("Checking v1alpha1 -> v1 conversion")
+			v1alpha1Profile.Name = "v1alpha"
+			v1alpha1Profile.ResourceVersion = ""
+			Expect(testclient.Client.Create(context.TODO(), v1alpha1Profile)).ToNot(HaveOccurred())
+
+			v1Profile := &performancev1.PerformanceProfile{}
+			key.Name = v1alpha1Profile.Name
+			err = testclient.GetWithRetry(context.TODO(), key, v1Profile)
+			ExpectWithOffset(1, err).ToNot(HaveOccurred(), "Failed getting v1profile")
+			Expect(verifyV1alpha1Conversion(v1alpha1Profile, v1Profile)).ToNot(HaveOccurred())
+
+			Expect(testclient.Client.Delete(context.TODO(), v1Profile)).ToNot(HaveOccurred())
+		})
+	})
 })
 
+func verifyV1alpha1Conversion(v1alpha1Profile *performancev1alpha1.PerformanceProfile, v1Profile *performancev1.PerformanceProfile) error {
+	specCPU := v1alpha1Profile.Spec.CPU
+	if (specCPU == nil) != (v1Profile.Spec.CPU == nil) {
+		return fmt.Errorf("spec CPUs field is different")
+	}
+
+	if specCPU != nil {
+		if string(*specCPU.Reserved) != string(*v1Profile.Spec.CPU.Reserved) {
```

Comment:
here and below: we check carefully if fields in the v1alpha1Profile are nil before to deference them, but it seems we are less careful with the v1Profile. Any reason for this?

Reply:
Please see the answer above. We are protected at this point, both have values or both have no value.